spec/lib/services/files-api-service-spec.js

```diff
@@ -0,0 +1,86 @@
+
+"use strict";
+
+describe("Http.Services.Api.Files", function () {
+    var skuService;
+    var filesApiService;
+    var waterline;
+    var workflowApiService;
+    var self = this;
+
+    before ("Http.Services.Api.Files before", function() {
+        helper.setupInjector([
+            dihelper.simpleWrapper({}, 'TaskGraph.Store'),
+            dihelper.simpleWrapper({}, 'TaskGraph.TaskGraph'),
+            helper.require("/lib/services/files-api-service.js"),
+            helper.require("/lib/services/sku-pack-service"),
+            helper.require("/lib/services/workflow-api-service"),
+            dihelper.requireWrapper('os-tmpdir', 'osTmpdir', undefined, __dirname),
+            dihelper.simpleWrapper(function() { arguments[1](); }, 'rimraf'),
+        ]);
+        filesApiService = helper.injector.get("Http.Services.Api.StaticFiles");
+        skuService = helper.injector.get('Http.Services.SkuPack');
+        waterline = helper.injector.get('Services.Waterline');
+        workflowApiService = helper.injector.get('Http.Services.Api.Workflows');
+
+        waterline.skus = {
+            findOne: function() {},
+        };
+
+        self.sandbox = sinon.sandbox.create();
+        self.sandbox.stub(waterline.skus, 'findOne').resolves('name');
+    });
+
+
+    beforeEach("Http.Services.Api.Files beforeEach", function() {
+        skuService.skuHandlers = {};
+        self.sandbox.reset();
+    });
+
+
+    describe("getAllStaticFiles", function() {
+
+        before(function(){
+            sinon.stub(skuService, 'regenerateSkus');
+        });
+        after(function(){
+            skuService.regenerateSkus.restore();
+        });
+
+        it('should expose the appropriate methods', function() {
+            filesApiService.should.have.property('getAllStaticFiles')
+            .that.is.a('function').with.length(0);
+        });
+
+        it('Run getAllStaticFiles', function() {
+            var sku = 
+            {   
+                "name": "my test sku",
+                "rules": [
+                    {   
+                        "path": "ipmi-fru.Builtin FRU Device (ID 0).Board Mfg",
+                        "contains": "DELL"
+                    },  
+                    {   
+                        "path": "ipmi-fru.Builtin FRU Device (ID 0).Board Part Number",
+                        "contains": "0CNCJW"
+                    }   
+                ],  
+                "taskRoot": "tasks",
+                "workflowRoot": "workflows",
+                "httpProfileRoot": "profiles",
+                "httpTemplatesRoot": "templates",
+                "httpStaticRoot": "static"
+            };  
+            waterline.skus.findOne.withArgs({name: "my test sku"}).resolves(sku);
+            skuService.regenerateSkus.resolves();
+            sinon.spy(filesApiService, "getAllStaticFiles");
+            filesApiService.getAllStaticFiles();
+            filesApiService.getAllStaticFiles.called.should.be.true;
```

Comment:
This doesn't appear to really be testing anything other than that a call to `getAllStaticFiles` doesn't throw an exception. It would be better to provide several test cases that exercise your code more fully, such as (at a minimum):

- The walker adds `sku` attributes to directories within skupack.d, as well as constructing result objects with a `uri` attribute like you expect (it's also good just so we have an example of the expected output in the test to look at).
- getAllStaticFiles handles ENOENT errors properly.
- getAllStaticFiles returns the expected flattenedPaths output data.
- pairSkuPackIds adds sku names to the result data.

I'm happy to show you ways in which you can create mocks for the walker, fs calls, etc. if you need examples.